backends/firrtl/firrtl.cc

```diff
@@ -392,7 +392,37 @@ struct FirrtlWorker
 		return result;
 	}
 
-	void run()
+	void emit_extmodule()
+	{
+		std::string moduleFileinfo = getFileinfo(module);
+		f << stringf("  extmodule %s: %s\n", make_id(module->name), moduleFileinfo.c_str());
+		vector<string> port_decls;
+
+		for (auto wire : module->wires())
+		{
+			const auto wireName = make_id(wire->name);
+			std::string wireFileinfo = getFileinfo(wire);
+
+			// Maybe not needed?
+			if (wire->port_id)
+			{
+				if (wire->port_input && wire->port_output)
```

Comment:
AFAIK if `port_input` or `port_output` are set you don't need to check `port_id`.

Reply:
Good to know. Thanks!